### DISCUSSION TITLE
Context propagation between microservices

### DISCUSSION THREAD

**Discussion by isma3ilovich** [open-telemetry/opentelemetry-cpp#614] [Q&A]
Hello,

I need to propagate context between two microservices

I use the function bellow to inject my  context in the first microservice

`std::string  PwcOtelCommonTracer::inject( nostd::shared_ptr<trace::Span> &rootSpan)
{
   std::map<std::string, std::string> headers;

   opentelemetry::trace::Scope scope(rootSpan);
   propagatorFormat.Inject(Setter, headers,opentelemetry::context::RuntimeContext::GetCurrent());

   return headers["traceparent"];
}`

 The function bellow to extract the context  in the second microservice: 

`void  PwcOtelCommonTracer::extract( std::string headers)
{
   std::map<std::string, std::string> carrier;
   carrier[std::string("traceparent")] = std::string(headers);

    auto current_ctx = opentelemetry::context::RuntimeContext::GetCurrent();
    auto ctx         = propagatorFormat.Extract(Getter, carrier, current_ctx);
    auto token       = opentelemetry::context::RuntimeContext::Attach(ctx);

}`

but when i create new span   it isn't attached with the propagated context

` auto  rootSpan=getTracer()->GetCurrentSpan();
 trace::Scope scope(rootSpan);
 getTracer()->StartSpan(name);`
 
Could you help me please ?

Thanks .

**Comment by lalitb** [comment]
@isma3ilovich  - Would you mind raising ticket for this, will take a look into it.

**Comment by isma3ilovich** [comment]
@lalitb  thank you for your reply .
I slove the problem by using 

`
    auto current_ctx = opentelemetry::context::RuntimeContext::GetCurrent();
    auto ctx         = propagatorFormat.Extract(Getter, carrier, current_ctx);
    /*auto token       = opentelemetry::context::RuntimeContext::Attach(ctx);*/
    auto ctx2_span        = ctx.GetValue(trace::kSpanKey);
    auto rootSpan             = nostd::get<nostd::shared_ptr<trace::Span>>(ctx2_span);
    return rootSpan;`
